careers/software-engineer.md

```diff
@@ -44,6 +73,21 @@ An engineer II is capable of taking on most technical problems and coordinating
 
 **Collaboration:** They are mindful of the work that others are doing and can critique or suggest improvements to the proposed plans of others. They get help from others when stuck, but are also capable of helping unstick their peers. They are capable of effectively running small to medium-sized projects with regular coaching.
 
+### Responsibilities
+
+* Design and develop software components or features with moderate complexity.
+* Write clean, efficient, and well-documented code with less direct supervision.
```

Comment:
Larger and optional thought: there and other places in this document, I'd like to find a different framing rather than "with x amount of supervision". 

Instead of framing it in terms of the amount of supervision, we should re-frame it terms of how the **engineer's** capabilities evolve at each level.


For this one, I'd reframe it as:

> Take ownership of code quality, making independent decisions about architecture, implementation, and documentation that align with team standards.